src/browser/setupWorker/setupWorker.ts

```diff
@@ -177,6 +178,14 @@ export class SetupWorkerApi
       options,
     ) as SetupWorkerInternalContext['startOptions']
 
+    invariant(
+      !this.context.startOptions.dangerouslyRunInProduction && isProduction(),
+      devUtils.formatMessage(
+        'The flag dangerouslyRunInProduction is false but you are in a production environment',
```

Comment:
What do you think if we phrase this error a bit better? Perhaps something like this?

- Failed to call "setupWorker" in a production environment. Please make sure you enable API mocking conditionally so it doesn't leak to production. 

Don't even mention the flag that controls it to emphasize the importance of this behavior. Still include a link to the docs about the flag for those who know what they are doing and want to suppress this behavior.

Reply:
@kettanaito let me know the link when you finish it